### PR TITLE
Route selector

### PR DESCRIPTION
**Pull request by Julias0**

### CODE REVIEW

**Comment by Julias0** [comment]
Screenshot 2021-07-12 at 5 43 37 PM
Screenshot 2021-07-12 at 5 44 29 PM

**Comment by Julias0** [comment]
Screenshot 2021-07-12 at 6 04 38 PM
Screenshot 2021-07-12 at 6 05 25 PM

File: src/app/shared/components/route-selector/route-selector.component.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  templateUrl: './route-selector.component.html',
+  styleUrls: ['./route-selector.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      multi: true,
+      useExisting: RouteSelectorComponent
+    },
+    {
+      provide: NG_VALIDATORS,
```
Comment (Julias0) [inline src/app/shared/components/route-selector/route-selector.component.ts:21]: This is to flow out validations from subforms to the main form.

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 2]
```diff
@@ -412,7 +417,7 @@ export class AddEditMileagePage implements OnInit {
           mileageCategoriesContainer: this.getMileageCategories()
         }).pipe(
           switchMap(({ expenseFieldsMap, mileageCategoriesContainer }) => {
-            const fields = ['purpose', 'txn_dt', 'cost_center_id', 'distance'];
+            const fields = ['purpose', 'txn_dt', 'cost_center_id'];
```
Comment (abhishek1234321) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:420]: why remove `distance`?

**Review by abhishek1234321 — APPROVED** [review]

**Pull request closed without merging**